snooty/semanticparser.py

```diff
@@ -9,12 +11,58 @@ def __init__(self) -> None:
 
     def run(self, pages: Dict[FileId, Page]) -> Dict[str, SerializableType]:
         # Specify which transformations should be included in semantic postprocessing
-        functions: List[
-            Callable[[Dict[FileId, Page]], Dict[str, SerializableType]]
-        ] = []
+        functions: List[Callable[[Dict[FileId, Page]], Dict[str, SerializableType]]] = [
+            self.slug_title
+        ]
         document: Dict[str, SerializableType] = {}
 
         for fn in functions:
             field: Dict[str, SerializableType] = fn(pages)
             document.update(field)
         return document
+
+    def slug_title(self, pages: Dict[FileId, Page]) -> Dict[str, SerializableType]:
+        # Function which returns a dictionary of slug-title mappings
+        slug_title_dict: Dict[str, SerializableType] = {}
+        for file_id in pages:
+            page = pages[file_id]
+            slug = str(page.source_path)
+
+            # Get the relative path from the absolute paths of the current file and the page
+            common_prefix = os.path.commonprefix([page.source_path, os.getcwd()])
+            if common_prefix != "":
+                slug = os.path.relpath(page.source_path, common_prefix)
+
+                # remove the file extension from the slug
+                ext_idx = slug.find(".")
+                slug = slug[:ext_idx]
+
+            # Skip slug-title mapping if the file is an `includes`
+            if "includes" in slug:
```

Comment:
We also don't want to include any files with "images" in the slug (for obvious reasons 😄).